x-pack/legacy/plugins/reporting/export_types/printable_pdf/server/execute_job/index.ts

```diff
@@ -18,44 +19,39 @@ import {
   getCustomLogo,
 } from '../../../common/execute_job/';
 
-export function executeJobFactory(server) {
+export const executeJobFactory: ExecuteJobFactory = function executeJobFactoryFn(
+  server: ServerFacade
+) {
   const generatePdfObservable = generatePdfObservableFactory(server);
   const logger = LevelLogger.createForServer(server, [PLUGIN_ID, PDF_JOB_TYPE, 'execute']);
 
-  return function executeJob(jobId, jobToExecute, cancellationToken) {
+  return function executeJob(
+    jobId: string,
+    jobToExecute: JobDocPayloadPDF,
+    cancellationToken: any
+  ) {
     const jobLogger = logger.clone([jobId]);
 
-    const process$ = Rx.of({ job: jobToExecute, server }).pipe(
+    const process$ = Rx.of({ job: jobToExecute, server, logger }).pipe(
       mergeMap(decryptJobHeaders),
-      catchError(err => {
-        jobLogger.error(err);
-        return Rx.throwError(
-          i18n.translate(
-            'xpack.reporting.exportTypes.printablePdf.compShim.failedToDecryptReportJobDataErrorMessage',
-            {
-              defaultMessage:
-                'Failed to decrypt report job data. Please ensure that {encryptionKey} is set and re-generate this report. {err}',
-              values: { encryptionKey: 'xpack.reporting.encryptionKey', err: err.toString() },
-            }
-          )
-        );
-      }),
```

Comment:
Logging the error of being unable to decrypt the job didn't feel like it belongs here. I moved that to the decrypt module